### ISSUE TITLE
price amount in API request and response

### ISSUE THREAD

**Issue by lucas03** [zalando/connexion#782]
### Description

How are you representing price amount in your APIs? I think most popular choices are strings and number. But for connexion, `type: number, format: decimal` is not supported yet I believe. What is the best practice for this?

In case of number, I guess there could be added some converter to convert number to decimal right after validation? What about returning price objects? Would it be also easy to serialize decimal for responses?

Strings I think are easier for handling, would be converted to decimal inside handler. But then there is missing validation and price is a number, not a string obviously.

I've seen this related issue in [OpenAPI](https://github.com/OAI/OpenAPI-Specification/issues/316#issuecomment-86594998)

**Comment by dtkav** [comment]
Hey @lucas03 - are you referring to this example in the cookbook docs?
https://github.com/zalando/connexion/blob/master/docs/cookbook.rst#custom-type-format

**Comment by lucas03** [comment]
I forgot I've read that in the past, thanks :) 

more or less I was wondering if you use string or float or custom type for money. I found a few related articles:

```
Fun fact: Int was actually suggestion by Mark Ralphson, member of OpenAPI initiative.

*Relevant texts to price in API*
https://stackoverflow.com/questions/35709595/why-would-you-use-a-string-in-json-to-represent-a-decimal-number
https://allegro-restapi-guideline.readthedocs.io/en/latest/Representation/#price-and-currency
https://www.oreilly.com/ideas/how-a-restful-api-represents-resources

*Usages:*

*String*
https://developer.paypal.com/docs/api/overview/#make-your-first-call
http://developers.payu.com/en/restapi.html
https://help.shopify.com/en/api/reference/sales-channels/payment
https://w3c.github.io/payment-request/
https://www.twilio.com/docs/phone-numbers/pricing
https://github.com/NationalBankBelgium/REST-API-Design-Guide/wiki/REST-Resources-Actions

*Number*
https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design
https://stripe.com/docs/api
there are more of course, but we should not decide based on which one is used the most.

*TLDR*: I’ve read a few articles and now I’m leaning towards string as well.

I’d use `{"type": "string", "format": "decimal"}` with custom validation.
```

thanks for linking me document, I'll resolve this issue now.

**Comment by dtkav** [comment]
Thanks for reporting back! Glad you were able to find a good solution, and
thanks for posting all that content for the next person to read up on.
